Sources/Brave/Frontend/Browser/Playlist/Managers & Cache/PlaylistCacheLoader.swift

```diff
@@ -330,49 +330,49 @@ class PlaylistWebLoader: UIView {
     self.removeFromSuperview()
   }
 
-  func load(url: URL, handler: @escaping (PlaylistInfo?) -> Void) {
-    self.handler = { [weak self] in
-      // Handler cannot be called more than once!
-      self?.handler = nil
-      handler($0)
-    }
-    
-    guard let webView = tab.webView,
-          let browserViewController = self.currentScene?.browserViewController else {
-      self.handler?(nil)
-      return
-    }
-    
-    self.certStore = browserViewController.profile.certStore
-    let KVOs: [KVOConstants] = [
-      .estimatedProgress, .loading, .canGoBack,
-      .canGoForward, .URL, .title,
-      .hasOnlySecureContent, .serverTrust,
-    ]
-
-    browserViewController.tab(tab, didCreateWebView: webView)
-    KVOs.forEach { webView.removeObserver(browserViewController, forKeyPath: $0.rawValue) }
-
-    // When creating a tab, TabManager automatically adds a uiDelegate
-    // This webView is invisible and we don't want any UI being handled.
-    webView.uiDelegate = nil
-    webView.navigationDelegate = self
-    
-    tab.replaceContentScript(PlaylistWebLoaderContentHelper(self),
-                             name: PlaylistWebLoaderContentHelper.scriptName,
-                             forTab: tab)
+  func load(url: URL) async -> PlaylistInfo? {
+    return await withCheckedContinuation { @MainActor continuation in
```

Comment:
`@MainActor` here won't do anything, since the body of a checked continuation is not an async context. If you need it to be on `MainActor` then you should give the annotation to the `load` method